app/assets/javascripts/dashboard/views/dataset/dataset_import_view.js

```diff
@@ -0,0 +1,157 @@
+// @flow
+
+import React from "react";
+import { Button, Spin, Input, Checkbox } from "antd";
+import Request from "libs/request";
+import update from "immutability-helper";
+import Toast from "libs/toast";
+import type { APIDatasetType } from "admin/api_flow_types";
+
+class DatasetImportView extends React.PureComponent {
+
+  props: {
+    datasetName: string,
+    isEditingMode: false,
+  }
+
+  state: {
+    dataset: ?APIDatasetType,
+    datasetJson: string,
+    isValidJSON: boolean,
+  } = {
+    dataset: null,
+    datasetJson: "",
+    isValidJSON: true,
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  async fetchData(): Promise<void> {
+    const datasetUrl = `/api/datasets/${this.props.datasetName}`;
+    const dataset = await Request.receiveJSON(datasetUrl);
+
+    const datasetJsonUrl = `${dataset.dataStore.url}/data/datasets/${this.props.datasetName}`;
+    const datasetJson = await Request.receiveJSON(datasetJsonUrl);
+
+    // eslint-disable-next-line react/no-did-mount-set-state
+    this.setState({
+      dataset,
+      datasetJson: JSON.stringify(datasetJson.dataSource, null, "  "),
+    });
+  }
+
+  importDataset = () => {
+    if (this.props.isEditingMode) {
+      const url = `/api/datasets/${this.props.datasetName}`;
+      Request.sendJSONReceiveJSON(url, {
+        data: this.state.dataset,
+      });
+    }
+
+    if (this.state.isValidJSON && this.state.dataset) {
+      const url = `${this.state.dataset.dataStore.url}/data/datasets/${this.props.datasetName}`;
+      Request.sendJSONReceiveJSON(url, {
+        data: JSON.parse(this.state.datasetJson),
+      }).then(() => {
+        Toast.success(`Successfully imported ${this.props.datasetName}`);
+        window.history.back();
+      });
+    } else {
+      Toast.error("Invalid JSON. Please fix the errors.");
+    }
+  }
+
+  handleChangeJson = (event: SyntheticInputEvent) => {
+    try {
+      JSON.parse(event.target.value);
+      this.setState({
+        datasetJson: event.target.value,
+        isValidJSON: true,
+      });
+    } catch (e) {
+      this.setState({
+        datasetJson: event.target.value,
+        isValidJSON: false,
+      });
+    }
+  }
+
+  handleChangeDescription = (event: SyntheticInputEvent) => {
+    this.updateDataset("description", event.target.value);
+  }
+
+  handleChangeCheckbox = (event: SyntheticInputEvent) => {
+    this.updateDataset("isPublic", event.target.checked);
+  }
+
+  updateDataset(propertyName: string, value: (string | boolean)) {
+    const newState = update(this.state, {
+      dataset: { [propertyName]: { $set: value } },
+    });
+    this.setState(newState);
```

Comment:
I think the method can also be written as
`this.setState({[propertyName]: value});`

`setState` doesn't need a complete new state. It will take care of updating only the existing keys. In fact, I think this code can be inlined. So `handleChangeDescription` would be `this.setState({description: event.target.value})`. Same goes for `handleChangeCheckbox`.